PROTOCOL.md

```diff
@@ -665,19 +566,103 @@ Field    | Type | Description
 `timestamp` | `number` | Timestamp of the `StreamMessage` published on the same stream and same partition by the same producer.
 `sequenceNumber` | `number` | Sequence Number of the `StreamMessage`.
 
-### GroupKey
 
-Used in key rotation to deliver a new key to existing subscribers. The format is as follows:
+## Tracker Protocol
+
+This document describes version `1` of the Tracker Control protocol.
+
+All messages are transmitted as JSON arrays and have the following structure with certain fields common to all messages: `[version, type, requestId, ...typeSpecificFields]`. 
+
+Field       | Type      | Description
+----------- | --------- | -----------
+`version`   | `number`  | The version of the Tracker Protocol
+`type`      | `number`  | An integer identifying the message type (see the table below)
+`requestId` |  `string` | Identifies a message. The `requestId` can be chosen freely for each request, as long as it's unique within the context of a particular connection. The same `requestId` value gets echoed back in response messages that result from the request.
+
+The `...typeSpecificFields` means that there are additional fields depending on message type. The possible message `type`s are:
+
+type | Description
+----------- | -----------
+1 | StatusMessage
+2 | InstructionMessage
+3 | StorageNodesRequest
+4 | StorageNodesResponse
+
+
+The individual types are described in the remainder of this section. We start by describing the requests and then the responses.
+
+These messages are exchanged between trackers and nodes for peer discovery and topology formation.
+
+### StatusMessage
+
+A node reports its status to the tracker using this message.
 
 ```
-[groupKeyId, encryptedGroupKey]
+[version, type, status]
 ```
 Example:
 ```
-["newGroupKeyId", "abc123"]
-``` 
+[2, 14, {
+    streams: [],
+    started: 0,
+    rtts: {}
+}]
+```
+
+Field    | Type | Description
+-------- | ---- | --------
+`status` | `object` | state of node
```

Comment:
Is the protocol opinionated on what the content of `status` should be? If yes, it should be defined here. The doc can also define it as flexible/arbitrary, but it should be defined one way or the other.

Reply:
The contents of `status` has changed rapidly compared to the rest of the protocol. The contents as they currently exist  also require further refining. Thus I left this undefined, with the intention of filling in the details later when the dust settles.

Reply:
Ok by me, but worth writing that down in the doc? (that the state object is subject to change)